earth2studio/models/px/dlesym.py

```diff
@@ -1060,7 +1106,89 @@ def coords_to_ll(self, coords: CoordSystem) -> CoordSystem:
             ll_coords.move_to_end(dim)
         return ll_coords
 
-    # @batch_func()
+    def _nan_interpolate_sst(
+        self, sst: torch.Tensor, coords: CoordSystem
+    ) -> torch.Tensor:
+        """
+        Custom interpolation to fill NaNs over landmasses in SST data.
+        """
+
+        da_sst = xr.DataArray(sst.cpu().numpy(), dims=coords.keys())
+        da_interp = da_sst.interpolate_na(
+            dim="lon", method="linear", use_coordinate=False
+        )
+
+        # Second pass: roll, interpolate along longitude, and unroll
```

Comment:
I understand why you roll, but would the interpolate in the previous step fill in some in accurate values on the 0 / 359.75 lons? Or does it leave the edges NaNs?

I.e. does the first step on something like [NaN, NaN, 1, 2, NaN, 3, NaN]

Give [NaN, NaN, 1, 2, 2.5, 3, NaN] or [1, 1, 1, 2, 2.5, 3, 3]